packages/@aws-cdk/aws-cloudfront/lib/edge-function.ts

```diff
@@ -0,0 +1,271 @@
+    const role = props.role ?? defaultLambdaRole(this, id);
+    const edgeFunction = new lambda.Function(this, 'Fn', {
+      ...props,
+      role,
+    });
+
+    return { edgeFunction, edgeArn: edgeFunction.currentVersion.edgeArn, functionStack: this.stack };
+  }
+
+  /** Create a support stack and function in us-east-1, and a SSM reader in-region */
+  private createCrossRegionFunction(id: string, props: lambda.FunctionProps): FunctionConfig {
+    const parameterNamePrefix = 'EdgeFunctionArn';
+    const parameterName = `${parameterNamePrefix}${id}`;
+    const functionStack = this.edgeStack();
+    this.stack.addDependency(functionStack);
+
+    const edgeFunction = functionStack.addEdgeFunction(id, parameterName, props);
+    // This is used to determine when the function has changed, to refresh the ARN from the custom resource.
+    const edgeFunctionHash = calculateFunctionHash(edgeFunction);
+
+    // Prefix of the parameter ARN that applies to all EdgeFunctions.
+    // This is necessary because the `CustomResourceProvider` is a singleton, and the `policyStatement`
+    // must work for multiple EdgeFunctions.
+    const parameterArnPrefix = this.stack.formatArn({
+      service: 'ssm',
+      region: EdgeFunctionExperimental.EDGE_REGION,
+      resource: 'parameter',
+      resourceName: parameterNamePrefix + '*',
+    });
+
+    const resourceType = 'Custom::CrossRegionStringParameterReader';
+    const serviceToken = CustomResourceProvider.getOrCreate(this, resourceType, {
+      codeDirectory: path.join(__dirname, 'edge-function'),
+      runtime: CustomResourceProviderRuntime.NODEJS_12,
+      policyStatements: [{
+        Effect: 'Allow',
+        Resource: parameterArnPrefix,
+        Action: ['ssm:GetParameter'],
+      }],
+    });
+    const resource = new CustomResource(this, 'ArnReader', {
+      resourceType: resourceType,
+      serviceToken,
+      properties: {
+        Region: EdgeFunctionExperimental.EDGE_REGION,
+        ParameterName: parameterName,
+        RefreshToken: edgeFunctionHash,
+      },
+    });
+    const edgeArn = resource.getAttString('FunctionArn');
+
+    return { edgeFunction, edgeArn, functionStack };
+  }
+
+  private edgeStack(): CrossRegionLambdaStack {
+    const stage = this.node.root;
+    if (!stage || !Stage.isStage(stage)) {
+      throw new Error('stacks which use EdgeFunctions must be part of a CDK app or stage');
+    }
+    const region = this.env.region;
+    if (Token.isUnresolved(region)) {
+      throw new Error('stacks which use EdgeFunctions must have an explicitly set region');
+    }
+
+    const edgeStackId = `edge-lambda-stack-${region}`;
+    let edgeStack = stage.node.tryFindChild(edgeStackId) as CrossRegionLambdaStack;
+    if (!edgeStack) {
+      edgeStack = new CrossRegionLambdaStack(stage, edgeStackId, {
+        synthesizer: this.crossRegionSupportSynthesizer(),
+        env: { region: EdgeFunctionExperimental.EDGE_REGION },
```

Comment:
Feels like these two settings should be baked into `CrossRegionLambdaStack`, so this could just be `new CrossRegionLambdaStack()`, no?